base_layer/core/src/base_node/state_machine_service/states/horizon_state_sync/horizon_state_synchronization.rs

```diff
@@ -524,6 +525,15 @@ impl<'a, B: BlockchainBackend + 'static> HorizonStateSynchronization<'a, B> {
             target: LOG_TARGET,
             "Horizon state validation succeeded! Committing horizon state."
         );
+        if syncing_header_hash !=
```

Comment:
This check should be moved to inside the `write_transaction`. There is still a chance that another thread can change the database in between this check and the `write_transaction` starting